src/main/java/org/takes/misc/Href.java

```diff
@@ -321,4 +292,53 @@ private static URI createURI(final String txt) {
         }
         return result;
     }
+
+    /**
+     * Convert the provided query into a Map.
+     * @param query The query to parse.
+     * @return A map containing all the query arguments and their values.
+     */
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+    private static SortedMap<String, List<String>> asMap(final String query) {
+        final SortedMap<String, List<String>> params =
+            new TreeMap<String, List<String>>();
+        if (query != null) {
+            final String[] pairs = query.split("&");
```

Comment:
@dmzaytsev I'm not sure I understand what you mean, could you please reformulate?